pwm/env_setup.sls

```diff
@@ -0,0 +1,157 @@
+setup_emailfrompwm:
+  file.managed:
+    - name: /usr/local/bin/emailfrompwm
+    - contents: 
+        - {{ pillar['emailfrompwm'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_emailfrom: 
+  file.managed:
+    - name: /usr/local/bin/emailfrom
+    - contents: 
+        - {{ pillar['emailfrom'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_adminemailto: 
+  file.managed:
+    - name: /usr/local/bin/adminemailto
+    - contents: 
+        - {{ pillar['adminemailto'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_ldaphostname: 
+  file.managed:
+    - name: /usr/local/bin/ldaphostname
+    - contents: 
+        - {{ pillar['ldaphostname'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_mailfromdomain: 
+  file.managed:
+    - name: /usr/local/bin/mailfromdomain
+    - contents: 
+        - {{ pillar['mailfromdomain'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_mailtoaddress: 
+  file.managed:
+    - name: /usr/local/bin/mailtoaddress
+    - contents: 
+        - {{ pillar['mailtoaddress'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_pwmurl: 
+  file.managed:
+    - name: /usr/local/bin/pwmurl
+    - contents: 
+        - {{ pillar['pwmurl'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_ostemail: 
+  file.managed:
+    - name: /usr/local/bin/ostemail
+    - contents: 
+        - {{ pillar['ostemail'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_osturl: 
+  file.managed:
+    - name: /usr/local/bin/osturl
+    - contents: 
+        - {{ pillar['osturl'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_ostapikey: 
+  file.managed:
+    - name: /usr/local/bin/ostapikey
+    - contents: 
+        - {{ pillar['ostapikey'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_oupath: 
+  file.managed:
+    - name: /usr/local/bin/oupath
+    - contents: 
+        - {{ pillar['oupath'] }}
+    - mode: 600
+    - user: root
+    - group: root
+
+setup_svcacctpass: 
+  file.managed:
+    - name: /usr/local/bin/svcacctpass
+    - contents: 
+        - {{ pillar['svcacctpass'] }}
```

Comment:
Is there a way to ensure the quoting along with the string value provided in the pillar data makes it into here so that salt doesn't bomb when special characters are used in the pillar data?

Reply:
Is "quoting" needed? I can try to set special characters and see what happens. At this moment I am unsure.